flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -1278,7 +1278,11 @@ object ScalarOperatorGens {
       val falseAction = generateIfElse(ctx, operands, resultType, i + 2)
 
       val Seq(resultTerm, nullTerm) = newNames("result", "isNull")
-      val resultTypeTerm = primitiveTypeTermForType(resultType)
+      val resultTypeTerm = if (trueAction.resultType.isNullable || falseAction.resultType.isNullable) {
+        boxedTypeTermForType(resultType)
+      } else {
+        primitiveTypeTermForType(resultType)
+      }
 
       val operatorCode = if (ctx.nullCheck) {
         s"""
```

Comment:
I think the reason is here, should first check `nullTerm`, if it is true, should not assign `resultTerm`.

Reply:
Thanks for the review.  The result for NPE is that we assign `null` to a primitive type variable here, if we not assign `resultTerm` when `nullTerm` is `true`, then `resultTerm` will not be `null` at last. IMHO, this should be `null` for `resultTerm`.

Or we can make `${trueAction.resultTerm}` be the `resultTerm` for the returned `GeneratedExpression` ?

Reply:
Ignore my above comment...
I've got your point, and will update the code very soon.